yivi-docs/blog/2025-09-05-ux-challenges/index.md

```diff
@@ -0,0 +1,117 @@
+
+Having both an expiry date and a maximum usage limit can be confusing for users. For example, if a user has a credential that expires in one year but can only be used five times, they may not understand why they cannot use it after five uses, even though it is still valid. The concept of maximum usage is something a user should not be bothered with, but in the current state of the art, it is unavoidable.
+
+<div class="center-container">
+    <img src="/img/batch-issuance.png" class="ss" alt="Yivi app UX design for batch issuance" />
+</div>
+<p style={{ textAlign: 'center', marginTop: '1em' }}>
+    Yivi app UX design for batch issuance
+</p>
+
+## Multiple credential formats
+Another challenge is that different services may require different credential formats. For example, some services may require SD-JWTs, while others may require W3C VCs, or even ISO mdocs. This creates a fragmented landscape that puts the burden on the end-user. From their perspective, a credential is a credential, they do not (and should not) need to understand which cryptographic format is being used under the hood.
+
+Without careful UX design, users could easily run into situations where:
+- They receive seemingly duplicate credentials that serve the same purpose but in different formats.
+- They are asked to re-collect the "same" credential just because a service does not accept the format they already have.
+- They are left wondering why a credential works in one context but fails in another.
+
+This can quickly lead to frustration, especially in scenarios where citizens are expected to interact with multiple services from different providers.
+
+Our approach in Yivi is to **embrace crypto agility**:  
+- Whenever possible, we will issue credentials in **all supported formats** at once.  
+- The wallet will abstract away the complexity, presenting users with a single, unified credential view.  
+- Behind the scenes, Yivi ensures that the correct format is presented to the relying party, without requiring user intervention.  
+
+To further support clarity, the wallet will offer **visual indicators** so that advanced users can see which credential type is being used (Idemix, SD-JWT, W3C VC, or mdoc). However, for most users, the complexity will remain hidden, allowing them to simply “use their credential” without worrying about formats or cryptography.
+
+:::warning
+We urge the community to adopt the same approach, to reduce the complexity for users.
+:::
+
+:::tip
+This design is not final and may change in the future. We welcome feedback and suggestions from the community.
+:::
+
+Below an image with visual indicators, showing the privacy first Idemix credential, SD-JWTs with a maximum usage, and the ISO mdoc credentials with a maximum usage.
+
+<div class="center-container">
+    <img src="/img/credential-formats.png" class="ss" alt="Yivi app retrieval wizard" />

```

Comment:
This is not the design we'll be adopting (for now).
As you mentioned, we don't want to burden the user with these details.
Maybe we can add this in the future (if there is a desire for them).
I'd ask the community to provide feedback on the difference between both UIs instead.

Reply:
I will make more clearly that this is an example design of where users end up with. And that we are gathering feedback in the community.